### PR TITLE
Misc fix

### PR DESCRIPTION
**Pull request by kroosec**

### CODE REVIEW

File: src/pluginscheduler.c [hunk 1]
```diff
@@ -406,8 +406,6 @@ scheduler_phase_cleanup (plugins_scheduler_t sched, int start, int end)
           plugin->deps = NULL;
           element = element->next;
         }
-      g_slist_free (sched->list[category]);
```
Comment (mattmundell) [inline src/pluginscheduler.c]: I don't have the internals knowledge to judge this one.

**Comment by mattmundell** [comment]
Besides the 2 I mentioned, all the changes are self contained and look correct.

File: nasl/nasl_text_utils.c [hunk 2]
```diff
@@ -271,7 +271,6 @@ nasl_strlen (lex_ctxt * lexic)
   tree_cell *retc;
 
   retc = alloc_tree_cell ();
-  retc->ref_count = 1;
```
[the remark this answers is not in the document]
Comment (kroosec) [reply nasl/nasl_text_utils.c]: ref_count always starts at zero (see other alloc_tree_cell()), deref_cell() when called frees the memory if decrementing it makes the value less than zero.

For reference, Valgrind output:

==4989== 56 bytes in 1 blocks are definitely lost in loss record 377 of 880
==4989==    at 0x4C31B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==4989==    by 0x5322A80: g_malloc0 (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.5600.3)
==4989==    by 0x4E98509: alloc_tree_cell (nasl_tree.c:38)
==4989==    by 0x4E95C85: nasl_strlen (nasl_text_utils.c:273)
==4989==    by 0x4E7C6EA: nasl_func_call (nasl_func.c:196)
==4989==    by 0x4E562DF: nasl_exec (exec.c:1024)
==4989==    by 0x4E53FE6: cell2atom (exec.c:221)
==4989==    by 0x4E54087: cell_cmp (exec.c:242)
==4989==    by 0x4E5806F: nasl_exec (exec.c:1586)
==4989==    by 0x4E53BB9: cell2bool (exec.c:80)
==4989==    by 0x4E56C23: nasl_exec (exec.c:1186)
==4989==    by 0x4E55B88: nasl_exec (exec.c:822)

File: src/pluginscheduler.c (hunk 1 above)
Comment (kroosec) [reply src/pluginscheduler.c]: For reference, relevant Valgrind output:
==31648== 4,248 bytes in 177 blocks are definitely lost in loss record 624 of 683
==31648==    at 0x4C31B25: calloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==31648==    by 0x5322A80: g_malloc0 (in /usr/lib/x86_64-linux-gnu/libglib-2.0.so.0.5600.3)
==31648==    by 0x117A0A: plugin_add (pluginscheduler.c:89)
==31648==    by 0x117E97: plugins_scheduler_enable (pluginscheduler.c:206)
==31648==    by 0x11827B: plugins_scheduler_init (pluginscheduler.c:293)
==31648==    by 0x110983: attack_network (attack.c:1043)
==31648==    by 0x11485C: handle_client (openvassd.c:440)
==31648==    by 0x114A17: scanner_thread (openvassd.c:494)
==31648==    by 0x119391: create_process (processes.c:90)
==31648==    by 0x115041: main_loop (openvassd.c:712)
==31648==    by 0x115AF4: main (openvassd.c:1081)

**Pull request merged**